addon/models/term.js

```diff
@@ -78,21 +69,16 @@ export default Model.extend({
    * @type {Ember.computed}
    * @public
    */
-  allParentTitles: computed('parent.{title,allParentTitles.[]}', function() {
-    return new Promise(resolve => {
-      this.get('parent').then(parentTerm => {
-        let titles = [];
-        if(!parentTerm){
-          resolve(titles);
-        } else {
-          parentTerm.get('allParents').then(parents => {
-            titles = titles.concat(parents.mapBy('title'));
-            titles.push(this.get('parent.title'));
-            resolve(titles);
-          });
-        }
-      });
-    });
+  allParentTitles: computed('parent.{title,allParentTitles.[]}', async function() {
+    const parentTerm = await this.get('parent');
+    if(!parentTerm){
+      return [];
+    }
+
+    const parents = await parentTerm.get('allParents');
+    const titles = parents.mapBy('title');
+    titles.push(this.get('parent.title'));
```

Comment:
Not sure if it matters, but since it is already in scope shouldn't this be `titles.push(parentTerm.get('title'));`?

Reply:
good catch. handled.